File: School/cis4500/quick.lua

function partition(array, low, high)
    pivot = array[low]
    left = low
    for i = low + 1, high do
        if array[i] < pivot then
            left = left + 1
            array[i], array[left] = array[left], array[i]
        end
    end
    array[low], array[left] = array[left], array[low]
    return left
end

function quickSort(array, low, high)
    if (low < high) then
        pivot = partition(array, low, high)
        quickSort(array, low, pivot)
        quickSort(array, pivot + 1, high)
    end
end

A = {}
asize = io.read("*number")
for i = 1,asize do 
    A[i] = math.random(10000000)
end
stime = os.clock()
quickSort(A,1,#A)
print(os.clock()-stime)

File: School/cis4500/quick.py
import random
import time
def partition(array, low, high):
    pivot = array[low]
    left = low
    for i in range(low+1,high):
        if array[i] < pivot:
            left = left + 1
            array[i], array[left] = array[left], array[i]
    array[low], array[left] = array[left], array[low]
    return left

def quickSort(array,low,high):
    if (low < high):
        pivot = partition(array,low,high)
        quickSort(array,low,pivot)
        quickSort(array,pivot+1,high)
 
A = []

size = int(raw_input(""))
for i in range(0,size):
    A.append(random.randrange(0,10000000))
t = time.time()
quickSort(A,0,len(A))
print((time.time()-t))